packages/nocodb/src/lib/db/sql-data-mapper/lib/sql/helpers/getAst.ts

```diff
@@ -86,7 +86,8 @@ const getAst = async ({
             (!fields?.length || fields.includes(col.title)) &&
             value
           : fields?.length
-          ? fields.includes(col.title)
+          ? fields.includes(col.title) &&
+            (col.uidt !== UITypes.LinkToAnotherRecord || value)
```

Comment:
Can you briefly explain why this change is required?

Reply:
Got it, better to change as follows `fields.includes(col.title) && value`

```js
    return {
      ...(await obj),
      [col.title]:
        allowedCols && (!includePkByDefault || !col.pk)
          ? allowedCols[col.id] &&
            (!isSystemColumn(col) || view.show_system_fields) &&
            (!fields?.length || fields.includes(col.title)) &&
            value
          : fields?.length
          ? fields.includes(col.title) && value
          : value
    };
```

Reply:
seems to be working, I will make the changes.